README.md

```diff
@@ -108,6 +112,119 @@ async onRequestInit({ requestInit }) {
+        var json = await new StreamReader(HttpContext.Request.Body).ReadToEndAsync();
+        Console.WriteLine($"Payload: {json}");
+
+        // Retrieve the Svix headers
+        var svixId = Request.Headers["svix-id"].ToString();
+        var svixTimestamp = Request.Headers["svix-timestamp"].ToString();
+        var svixSignature = Request.Headers["svix-signature"].ToString();
+
+        Console.WriteLine($"Svix ID: {svixId}");
+        Console.WriteLine($"Svix Timestamp: {svixTimestamp}");
+        Console.WriteLine($"Svix Signature: {svixSignature}");
+
+        if (string.IsNullOrEmpty(svixId) || string.IsNullOrEmpty(svixTimestamp) || string.IsNullOrEmpty(svixSignature))
+        {
+            return BadRequest("Missing headers");
+        }
+
+        // Attempt signature verification with the raw signature
+        var wh = new Webhook(clerkWebhookSecret);
+        WebHeaderCollection headers = new WebHeaderCollection
+        {
+            { "svix-id", svixId },
+            { "svix-timestamp", svixTimestamp },
+            { "svix-signature", svixSignature }
+        };
+
+        Event webhookEvent;
+        try
+        {
+            wh.Verify(json, headers); // Verify doesn't return an event, just verifies the signature
+            webhookEvent = JsonConvert.DeserializeObject<Event>(json); // Deserialize the JSON into an Event object
+            Console.WriteLine($"Event Type: {webhookEvent.Type}");
+        }
+        catch (Svix.Exceptions.WebhookVerificationException ex)
+        {
+            return BadRequest("Invalid signature");
+        }
+        catch (Exception ex)
+        {
+            return BadRequest("An error occurred");
+        }
+
+        switch (webhookEvent.Type)
+        {
+            case "user.created":
+                // Handle user created event
+                break;
+            case "user.updated":
+                // Handle user updated event
+                break;
+            case "user.deleted":
+                // Handle user deleted event
+                break;
+
+            default:
+                return BadRequest("Unhandled event type");
+        }
+
+        return Ok();
+    }
+
+    // Define the Event class to represent the webhook event
+    public class Event
+    {
+        public string Type { get; set; } = string.Empty; // Initialize with an empty string
+        public ClerkUser Data { get; set; } = new ClerkUser(); // Initialize with a new instance
+    }
+
+    public class ClerkUser
+    {
+        public string Id { get; set; } = string.Empty;
+        public string ExternalId { get; set; } = string.Empty;
+        public string First_Name { get; set; } = string.Empty;
+        public string Last_Name { get; set; } = string.Empty;
+        public List<ClerkEmailAddress> Email_Addresses { get; set; } = new List<ClerkEmailAddress>();
+    }
+
+    public class ClerkEmailAddress
+    {
+        public string Email_Address { get; set; } = string.Empty;
+    }
```

Comment:
Properties shouldn't be initialized to string by default, show them as nullable or `null!` forgiving based on Clerk's user model. Use `PascalCase` & `JsonPropertyName` for underscored properties.

Reply:
Good points, thanks. Just started with C# this week so these pointers definitely help, appreciate it :)